### ISSUE TITLE
Podman remote tries SELinux relabel of incorrect directory, unless given an absolute path

### ISSUE THREAD

**Issue by 1player** [containers/podman#14528]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Podman tries to relabel the incorrect directory when starting a container in remote mode, unless given an absolute path to the volume.

**Steps to reproduce the issue:**

1. Set up a container with podman remote access. 
2. Enter the container.
3. `mkdir test && cd test`
4. `podman run -it -v .:/foo:z busybox sh`

**Describe the results you received:**

```
Error: error preparing container d41d9adeaad2a3984d1a181a7fec432172a3836d3736632dfd764705f20b90f2 for attach: SELinux relabeling of /var/home/user is not allowed
```

Note that podman is trying to relabel my whole home directory, not just `/var/home/user/test` as expected. 

Works correctly if I pass the absolute path of my volume instead of a relative one: `podman run -it -v $(pwd):/foo:z busybox sh`

**Describe the results you expected:**

`-v .:/foo` and `-v $(pwd):/foo` to be equivalent in remote mode.

**Additional information you deem important (e.g. issue happens only occasionally):**

Affects only podman --remote. 

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.1.0
API Version:  4.1.0
Go Version:   go1.18.1
Git Commit:   e4b03902052294d4f342a185bb54702ed5bed8b1
Built:        Fri May  6 19:18:30 2022
OS/Arch:      linux/amd64

Server:       Podman Engine
Version:      4.1.0
API Version:  4.1.0
Go Version:   go1.18
Built:        Fri May  6 17:15:54 2022
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.26.1
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 97.92
    systemPercent: 0.5
    userPercent: 1.59
  cpus: 20
  distribution:
    distribution: fedora
    variant: silverblue
    version: "36"
  eventLogger: journald
  hostname: tranquility
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.17.12-300.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 54868623360
  memTotal: 67339501568
  networkBackend: cni
  ociRuntime:
    name: crun
    package: crun-1.4.5-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.5
      commit: c381048530aa750495cf502ddb7181f2ded5b400
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 1h 35m 50.55s (Approximately 0.04 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /var/home/sph/.config/containers/storage.conf
  containerStore:
    number: 21
    paused: 0
    running: 2
    stopped: 19
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/sph/.local/share/containers/storage
  graphRootAllocated: 1998678130688
  graphRootUsed: 188508585984
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 15
  runRoot: /run/user/1000/containers
  volumePath: /home/sph/.local/share/containers/storage/volumes
version:
  APIVersion: 4.1.0
  Built: 1651853754
  BuiltTime: Fri May  6 17:15:54 2022
  GitCommit: ""
  GoVersion: go1.18
  Os: linux
  OsArch: linux/amd64
  Version: 4.1.0
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

Host: `podman-4.1.0-1.fc36.x86_64`
Remote: podman 4.1.0-1 on Arch Linux x86_64

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Comment by rhatdan** [comment]
Very strange. Does the . directory end up being correct if you do not specify :Z, IE does

podman run -it --privileged -v .:/foo busybox ls /foo

End up with your homedir or with the test directory?

**Comment by rhatdan** [comment]
@vrothberg @ashley-cui @umohnani8 This looks like a good one for an intern.

**Comment by 1player** [comment]
> Very strange. Does the . directory end up being correct if you do not specify :Z, IE does
> 
> podman run -it --privileged -v .:/foo busybox ls /foo
> 
> End up with your homedir or with the test directory?

That lists my home directory, even if my cwd is somewhere else, so that definitely is incorrect behaviour.

**Comment by 1player** [comment]
Still a valid bug. In remote mode podman volumes mount the wrong directory, which, again, sounds like a major bug.

**Comment by rhatdan** [comment]
Seems to be fixed in podman 4.2.

$ pwd
/tmp/test
$ ls -lZ
total 0
-rw-rw-r--. 1 dwalsh dwalsh system_u:object_r:user_tmp_t:s0 0 Jul 22 08:04 dan
-rw-rw-r--. 1 dwalsh dwalsh system_u:object_r:user_tmp_t:s0 0 Jul 22 08:04 walsh
$ ~/podman/bin/podman --remote run -v .:/mnt:Z fedora ls -lZ /mnt
total 0
-rw-rw-r--. 1 root root system_u:object_r:container_file_t:s0:c482,c813 0 Jul 22 08:04 dan
-rw-rw-r--. 1 root root system_u:object_r:container_file_t:s0:c482,c813 0 Jul 22 08:04 walsh
$ ls -lZ
total 0
-rw-rw-r--. 1 dwalsh dwalsh system_u:object_r:container_file_t:s0:c482,c813 0 Jul 22 08:04 dan
-rw-rw-r--. 1 dwalsh dwalsh system_u:object_r:container_file_t:s0:c482,c813 0 Jul 22 08:04 walsh
